app/main.py

```diff
@@ -1,3 +1,80 @@
+import json
+import math
+import datetime
+FUEL_PRICE = 2.4
+
+
 def shop_trip():
-    # write your code here
-    pass
+    with open("app/config.json") as json_input_file:
+        data = json.load(json_input_file)
+    shops = data["shops"]
+    for customer in data["customers"]:
+        riding_to_the_store(customer, shops)
+
+
+def get_distance(customer: dict, shop: dict):
+    distance = math.dist(customer["location"], shop["location"])
+    return distance
+
+
+def get_cost_of_ride_to_the_store(customer: dict, shop: dict):
+    distance = get_distance(customer, shop)
+    need_fuel = ((customer["car"]["fuel_consumption"] / 100) * distance)
+    fuel_cost = need_fuel * FUEL_PRICE
+    products_cost = 0
+    for keys, values in customer["product_cart"].items():
+        cost = values * shop["products"][keys]
+        products_cost += cost
+    return round(fuel_cost * 2 + products_cost, 2)
+
+
+def riding_to_the_store(customer: dict, shops: list):
+    name = customer["name"]
+    money = customer["money"]
+    print(f"{name} has {money} dollars")
+    cost_of_purchases = []
+    stores = {}
+    for shop in shops:
+        shop_name = shop["name"]
+        cost = get_cost_of_ride_to_the_store(customer, shop)
+        cost_of_purchases.append(cost)
+        stores[shop_name] = cost
+        print(f"{name}'s trip to the {shop_name} costs {cost}")
+    max_cost = max(cost_of_purchases)
+    cheapest_store = min(stores, key=stores.get)
+    for shop in shops:
+        if shop["name"] == cheapest_store:
+            cheapest_store = shop
+    if money >= max_cost:
```

Comment:
I'm not sure but if we don't have the money for the most expensive trip then we don't go at all? Will your program work when we have 10$ and trips costs 5, 8, 15 for example?

Reply:
You are right